src/SandboxJulia.jl

```diff
@@ -0,0 +1,180 @@
+Julia is installed can be chosen.
+"""
+function run_sandboxed_julia(args=``; wait=true,
+                             registries_dir=joinpath(first(DEPOT_PATH), "registries"),
+                             storage_dir=mktempdir(),
+                             install::String=dirname(dirname(Sys.which("julia"))),
+                             mounts::Dict{String,String}=Dict{String,String}(),
+                             privileged=false,
+                             kwargs...)
+    config, cmd = runner_sandboxed_julia(install, args; storage_dir, registries_dir, kwargs...)
+
+    # XXX: even when preferred_executor() returns UnprivilegedUserNamespacesExecutor,
+    #      sometimes a stray sudo happens at run time? no idea how.
+    exe_typ = privileged ? PrivilegedUserNamespacesExecutor : UnprivilegedUserNamespacesExecutor
+    exe = exe_typ()
+    proc = Base.run(exe, config, cmd; wait)
+
+    # TODO: introduce a --stats flag that has the sandbox trace and report on CPU, network, ... usage
+
+    if wait
+        cleanup(exe)
+    else
+        @async begin
+            try
+                Base.wait(proc)
+                cleanup(exe)
+            catch err
+                @error "Unexpected error while cleaning up process" exception=(err, catch_backtrace())
+            end
+        end
+    end
+
+    return proc
+end
+
+# global Xvfb process for use by all containers
+const xvfb_lock = ReentrantLock()
+const xvfb_proc = Ref{Union{Base.Process,Nothing}}(nothing)
+
+ 
+function installed_julia_dir(jp)
+    jp_contents = readdir(jp)
+    # Allow the unpacked directory to either be insider another directory (as produced by
+    # the buildbots) or directly inside the mapped directory (as produced by the BB script)
+    if length(jp_contents) == 1
+        jp = joinpath(jp, first(jp_contents))
+    end
+    jp
+end
+
+function runner_sandboxed_julia(install::String, args=``; registries_dir,
+                                storage_dir,
+                                dot_julia_path = mktempdir(),
+                                install_dir="/opt/julia",
+                                stdin=stdin, stdout=stdout, stderr=stderr,
+                                env::Dict{String,String}=Dict{String,String}(),
+                                mounts::Dict{String,String}=Dict{String,String}(),
+                                xvfb::Bool=true, cpus::Vector{Int}=Int[])
+    julia_path = installed_julia_dir(install)
+    rootfs = prepare_rootfs()
+    read_only_maps = Dict(
+        "/"                                 => rootfs.path,
+        install_dir                         => julia_path,
+        "/usr/local/share/julia/registries" => registries_dir,
+    )
+
+    artifacts_path = joinpath(storage_dir, "artifacts")
+    mkpath(artifacts_path)
+    read_write_maps = merge(mounts, Dict(
+        joinpath(rootfs.home, ".julia/artifacts")   => artifacts_path,
+        joinpath(rootfs.home, ".julia")   => dot_julia_path
```

Comment:
I'm not sure we want to map over the whole `.julia` but I would get permissions errors otherwise. It's not clear to me if `read-write` means the rootfs is read-only except for the write-mapped bits, or if it means these are the only directories that the code in the sandbox can write to on the host filesystem.